test/net_http_test.rb

```diff
@@ -0,0 +1,389 @@
+          Net::HTTP.start(uri.host, uri.port) {}
+        end
+
+        close_circuit!
+        Net::HTTP.start(uri.host, uri.port) do |http|
+          open_circuit!
+          get_subclasses(Net::HTTPRequest).each do |action|
+            assert_raises(Net::CircuitOpenError, "#{action.name} did not raise a Net::CircuitOpenError") do
+              request = action.new uri
+              http.request(request)
+            end
+          end
+        end
+      end
+    end
+  end
+
+  def test_custom_raw_semian_options_work
+    with_server do
+      semian_config = {}
+      semian_config["development"] = {}
+      semian_config["development"]["http_default"] =
+        {"tickets" => 1,
+         "success_threshold" => 1,
+         "error_threshold" => 3,
+         "error_timeout" => 10}
+      semian_config["development"]["http_localhost_#{TOXIC_PORT}"] =
+        {"tickets" => 1,
+         "success_threshold" => 1,
+         "error_threshold" => 3,
+         "error_timeout" => 10}
+      sample_env = "development"
+
+      semian_options_proc = proc do |semian_identifier|
+        if !semian_config[sample_env].key?(semian_identifier)
+          semian_config[sample_env]["http_default"]
+        else
+          semian_config[sample_env][semian_identifier]
+        end
+      end
+
+      with_semian_options(semian_options_proc) do
+        Net::HTTP.start("localhost", TOXIC_PORT) do |http|
+          assert_equal semian_config["development"][http.semian_identifier], http.raw_semian_options
+        end
+        Net::HTTP.start("localhost", PORT) do |http|
+          assert_equal semian_config["development"]["http_default"], http.raw_semian_options
+        end
+        assert_equal semian_config["development"]["http_default"],
+                     Semian::NetHTTP.retrieve_semian_options_by_identifier("http_default")
+      end
+    end
+  end
+
+  def test_custom_raw_semian_options_can_disable
+    with_server do # disabled if nil
+      semian_options_proc = proc { nil }
+      with_semian_options(semian_options_proc) do
+        http = Net::HTTP.new("localhost", "#{TOXIC_PORT}")
+        assert_equal false, http.enabled?
+      end
+    end
+
+    with_server do # disabled if key not found
+      semian_config = {}
+      semian_config["development"] = {}
+      semian_config["development"]["http_localhost_#{TOXIC_PORT}"] =
+        {"tickets" => 1,
+         "success_threshold" => 1,
+         "error_threshold" => 3,
+         "error_timeout" => 10}
+      sample_env = "development"
+
+      semian_options_proc = proc do |semian_identifier|
+        semian_config[sample_env][semian_identifier]
+      end
+      with_semian_options(semian_options_proc) do
+        http = Net::HTTP.new("localhost", "#{TOXIC_PORT}")
+        assert_equal true, http.enabled?
+
+        http = Net::HTTP.new("localhost", "#{TOXIC_PORT + 100}")
+        assert_equal false, http.enabled?
+      end
+    end
+  end
+
+  def test_adding_custom_errors_work
```

Comment:
You're testing here that adding custom errors append it to the list. You should have a test that actually asserts that when it's raised it's caught and Semian counts it towards the circuit breaker.